cmd/charm/charmcmd/publish.go

```diff
@@ -95,19 +97,23 @@ var publishCharm = func(client *csclient.Client, id *charm.URL, channels []param
 
 func (c *publishCommand) Run(ctxt *cmd.Context) error {
 	// Instantiate the charm store client.
-	client, err := newCharmStoreClient(ctxt, c.auth)
+	client, err := newCharmStoreClient(ctxt, c.auth, params.NoChannel)
 	if err != nil {
 		return errgo.Notef(err, "cannot create the charm store client")
 	}
 	defer client.jar.Save()
 
-	err = publishCharm(client.Client, c.id, []params.Channel{params.Channel(c.channel)}, c.resources)
+	channel := c.channel.C
+	if channel == params.NoChannel {
```

Comment:
Rather than doing this, we could instantiate the chanValue to have StableChannel as its value. I think you'll then see it as the default when using --help (worth checking that I'm correct there thought)